File: src/main/java/chapter01/Main.java
package chapter01;

public class Main {
  public static void main(String[] args) {
    Bookshelf bookshelf = new Bookshelf(4);
    bookshelf.appendBook(new Book("Around the World in 80 Days"));
    bookshelf.appendBook(new Book("Bible"));
    bookshelf.appendBook(new Book("Cinderella"));
    bookshelf.appendBook(new Book("Daddy-Long-Legs"));

    for (Iterator<Book> iter = bookshelf.iterator(); iter.hasNext();) {
      Book book = iter.next();
      System.out.println(book.getName());
    }
  }
}

File: src/main/scala/chapter01/Main.scala
package chapter01

case class Book(name: String)

class Bookshelf(books: Book*) extends Traversable[Book] {
  def foreach[U](f: Book => U) = books.foreach(f)
}

object Main {
  def main(args: Array[String]): Unit = {
    val bookshelf = new Bookshelf(
      Book("Around the World in 80 Days"),
      Book("Bible"),
      Book("Cinderella"),
      Book("Daddy-Long-Legs")
    )
    for (book <- bookshelf) println(book.name)
  }
}
